FirebaseCore/Tests/Unit/Swift/FirebaseAppTests.swift

```diff
@@ -0,0 +1,382 @@
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+  }
+
+  func testOptionsForApp() {
+    FirebaseApp.configure()
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let defaultApp = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+    let defaultOptions = FirebaseOptions.defaultOptions()
+    XCTAssertEqual(defaultApp.options, defaultOptions)
+
+    let options = FirebaseOptions(googleAppID: testOptions.googleAppID,
+                                  gcmSenderID: testOptions.gcmSenderID)
+    let superSecretURLScheme = "com.supersecret.googledeeplinkurl"
+    options.deepLinkURLScheme = superSecretURLScheme
+    FirebaseApp.configure(name: testApp.testAppName1, options: options)
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+    XCTAssertEqual(app.options.googleAppID, testOptions.googleAppID)
+    XCTAssertEqual(app.options.gcmSenderID, testOptions.gcmSenderID)
+    XCTAssertEqual(app.options.deepLinkURLScheme, superSecretURLScheme)
+    XCTAssertNil(app.options.androidClientID)
+  }
+
+  func testFirebaseDataCollectionDefaultEnabled() {
+    FirebaseApp.configure()
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+
+    // defaults to true unless otherwise set to no in app's Info.plist
+    XCTAssertTrue(app.isDataCollectionDefaultEnabled)
+
+    app.isDataCollectionDefaultEnabled = false
+    XCTAssertFalse(app.isDataCollectionDefaultEnabled)
+
+    // reset to defautl true since it will persist across runs of the app/tests
+    app.isDataCollectionDefaultEnabled = true
+  }
+
+  // MARK: - Helpers
+
+  private func expectedUserInfoForApp(named name: String, isDefaultApp: Bool) -> NSDictionary {
+    return [
+      "FIRAppNameKey": name,
+      "FIRAppIsDefaultAppKey": NSNumber(value: isDefaultApp),
+      "FIRGoogleAppIDKey": testOptions.googleAppID,
+    ]
+  }
+
+  private func expectAppConfigurationNotification(with expectedUserInfo: NSDictionary) {
+    expectation(forNotification: NSNotification.Name.firAppReadyToConfigureSDK,
+                object: FirebaseApp.self, handler: { (notification) -> Bool in
+                  if let userInfo = notification.userInfo {
+                    if expectedUserInfo.isEqual(to: userInfo) {
+                      return true
+                    }
+                  } else {
+                    XCTFail("Failed to unwrap notification user info")
+                  }
+                  return false
+    })
+  }
```

Comment:
> nit: I wonder if `expectAppConfigurationNotification` method can be used here as well? It seems like we may just call the method twice to create two separate expectations instead of using `appsConfiguredSuccessfullyExpectation`. Though I am not sure why `appsConfiguredSuccessfullyExpectation` can be fulfilled twice with the current implementation. I would double check if the tests is actually working as expected.

@maksymmalyhin Because configuring two apps will post notifications with the same notification name, I was running into issues where, when I called `expectAppConfigurationNotification` twice when configuring two apps, the first notification posted would attempt to trigger both expectations. Now, I return `true` (which fulfills the expectation) whenever we match with notification's `userInfo` to the `expectedUserInfo`. If one of the notifications doesn't post or the notification's `userInfo` doesn't match the expected, then the expectation won't be fulfilled and could eventually timeout and fail the test.